server/src/main/java/io/crate/expression/scalar/CurrentDateFunction.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.expression.scalar;
+
+import io.crate.data.Input;
+import io.crate.metadata.NodeContext;
+import io.crate.metadata.Scalar;
+import io.crate.metadata.TransactionContext;
+import io.crate.metadata.functions.Signature;
+import io.crate.types.DataTypes;
+
+
+public final class CurrentDateFunction extends Scalar<Long, String> {
+
+    public static final String NAME = "curdate";
+
+    public static void register(ScalarFunctionModule module) {
+        module.register(
+            Signature.scalar(
+                NAME,
+                DataTypes.DATE.getTypeSignature()
+            ).withFeatures(NO_FEATURES),
+            CurrentDateFunction::new
+        );
+    }
+
+    private final Signature signature;
+    private final Signature boundSignature;
+
+    public CurrentDateFunction(Signature signature, Signature boundSignature) {
+        this.signature = signature;
+        this.boundSignature = boundSignature;
+    }
+
+    @Override
+    @SafeVarargs
+    public final Long evaluate(TransactionContext txnCtx, NodeContext nodeCtx, Input<String>... args) {
+        return txnCtx.currentInstant().toEpochMilli();
```

Comment:
I think the value needs to be truncated to the date, no?

```
cr> select date_trunc('day', current_timestamp), current_date;
+--------------------------------------+---------------+
| date_trunc('day', CURRENT_TIMESTAMP) |     curdate() |
+--------------------------------------+---------------+
|                        1615766400000 | 1615800127721 |
+--------------------------------------+---------------+
```

Reply:
Thanks for spotting this, forgot to update logic when changed LocalDate to Long. Added a test for that case.